docs/integrations/data-ingestion/data-formats/json/schema.md

```diff
@@ -744,46 +776,190 @@ CREATE TABLE people
+
+1 row in set. Elapsed: 0.440 sec.
+```
+
+Note how our columns have excluded from our data:
+
+```sql
+
+SELECT *
+FROM people
+FORMAT PrettyJSONEachRow
 
-```json
 {
-  "id": 1,
-  "name": "Clicky McCliickHouse",
-  "username": "Clicky",
-  "email": "clicky@clickhouse.com",
-  "tags": [
-    {
-      "key": "hobby",
-      "name": "Diving",
-      "time": "2024-07-11 14:18:01"
-    },
-    {
-      "key": "car",
-      "name": "Tesla",
-      "time": "2024-07-11 15:18:23"
+    "json": {
+        "dob" : "1992-07-15",
+        "id" : "2",
+        "name" : "Analytica Rowe",
+        "phone_numbers" : [
+            "123-456-7890",
+            "555-867-5309"
+        ],
+        "tags" : {
+            "car" : {
+                "model" : "Audi e-tron",
+                "year" : "2022"
+            },
+            "hobby" : "Running simulations",
+            "holidays" : [
+                {
+                    "location" : "Kyoto, Japan",
+                    "year" : "2023"
+                }
+            ]
+        },
+        "username" : "Analytica",
+        "website" : "fastdata.io"
+    }
+}
+{
+    "json": {
+        "dob" : "2007-03-31",
+        "email" : "clicky@clickhouse.com",
+        "id" : "1",
+        "name" : "Clicky McCliickHouse",
+        "phone_numbers" : [
+            "010-692-6593",
+            "020-192-3333"
+        ],
+        "tags" : {
+            "car" : {
+                "model" : "Tesla",
+                "year" : "2023"
+            },
+            "hobby" : "Databases",
+            "holidays" : [
+                {
+                    "location" : "Azores, Portugal",
+                    "year" : "2024"
+                }
+            ]
+        },
+        "username" : "Clicky",
+        "website" : "clickhouse.com"
     }
-  ]
 }
+
+2 rows in set. Elapsed: 0.004 sec.
+```
+
+
+#### Optimizing performance with type hints {#optimizing-performance-with-type-hints}  
+
+Type hits offer more than just a way to avoid unnecessary type inference - they eliminate storage and processing indirection entirely, as well as allowing [optimal primitive types](/data-modeling/schema-design#optimizing-types) to be specified. JSON paths with type hints are always stored just like traditional columns, bypassing the need for [**discriminator columns**](https://clickhouse.com/blog/a-new-powerful-json-data-type-for-clickhouse#storage-extension-for-dynamically-changing-data) or dynamic resolution during query time. 
```

Comment:
```suggestion
Type hints offer more than just a way to avoid unnecessary type inference - they eliminate storage and processing indirection entirely, as well as allowing [optimal primitive types](/data-modeling/schema-design#optimizing-types) to be specified. JSON paths with type hints are always stored just like traditional columns, bypassing the need for [**discriminator columns**](https://clickhouse.com/blog/a-new-powerful-json-data-type-for-clickhouse#storage-extension-for-dynamically-changing-data) or dynamic resolution during query time. 
```